core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -266,19 +278,11 @@ private[spark] class TaskSetManager(
     taskAttempts(taskIndex).exists(_.host == host)
   }
 
-  /**
-   * Is this re-execution of a failed task on an executor it already failed in before
-   * EXECUTOR_TASK_BLACKLIST_TIMEOUT has elapsed ?
-   */
-  private[scheduler] def executorIsBlacklisted(execId: String, taskId: Int): Boolean = {
-    if (failedExecutors.contains(taskId)) {
-      val failed = failedExecutors.get(taskId).get
-
-      return failed.contains(execId) &&
-        clock.getTimeMillis() - failed.get(execId).get < EXECUTOR_TASK_BLACKLIST_TIMEOUT
-    }
-
-    false
+  private def blacklistedOnExec(execId: String, host: String, index: Int): Boolean = {
```

Comment:
can you call this taskIsBlacklistedOnExec? (to make it obvious this is task-specific)